maintenance/code-backports.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+# silence debug output from VCS
+beQuiet="--quiet"
+#beQuiet=""
+version=`basename "$1" | sed s/squid-//`
+vnext=$(( $version + 1 ))
+srcbranch=v$vnext
+
+cd $1
+
+# Dev branch does not receive backports
+test -f .BASE && exit 0
+
+# clean the current workspace
+gitCleanWorkspace ()
+{
+  git clean $beQuiet -xdf --exclude="\.BASE"
+  git checkout $beQuiet -- .
+}
+
+if ! test -d .git ; then
+  echo "ERROR: missing git repository"
+  exit 1;
+fi
+if ! test -d ../squid-$vnext ; then
+  echo "ERROR: missing git repository to backport from"
+  exit 1;
+fi
+if test -e ../squid-$vnext/.BASE ; then
+  srcbranch=`cat ../squid-$vnext/.BASE`
+fi
+
+echo "FOUDN srcbranch=$srcbranch"
+
+# Prepare backports branch for updates
+git checkout $beQuiet v$version-next-backports 2>/dev/null &&
+  gitCleanWorkspace &&
+    git rebase origin/v$version 2>/dev/null || ( git rebase --abort 2>/dev/null ; exit 1 )
+
+# handler on successful backport
+doneCherryPick ()
+{
+  sha="$1"
+  prnum="$2"
+  version="$3"
+
+  git push $beQuiet >/dev/null
+  # TODO automatically remove label from PRs on successful commit+push to vN-next-backports
```

Comment:
FWIW, I would consider leaving some PR label[^1] after merging the backported commit, as a clear sign of what has happened to the original PR -- an easily accessible documentation of where it has been backported. Yes, if the backported commit refers to the original PR by that PR number, then GitHub will link the two, but those links are often buried in GitHub page noise, and we should not really use GitHub PR numbers for such references anyway (we should use commit SHAs instead, but SHAs that do not generate PR links).

[^1]: Labeling the original PR means either leaving the original PR label in place or, better, replacing that with a new/dedicated label. For example, replacing `backport-to-v5` with `backported-to-v5` or something like that.

Reply:
Sans an agreed label to add. Using an automatic comment instead.

Reply:
The current "queued for backport to vN" comment does not address this change request because it reflects the desire to backport, not the fact of a successful backport. FWIW, a comment is also much more difficult to find, especially in a noisy PR or when searching across multiple PRs. I do support adding comments, but comments solve a different set of problems than labels do.

Obviously, we can find a suitable name for the new label. I am not aware of any relevant failures, and I do not even recall any serious attempts to do that beyond my `backported-to-vN` suggestion in this change request, but we can revisit them if I have forgotten about them.